File: src/ZenMoney/ZenMoney.Application/Models/Expense/ExpenseModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ZenMoney.Application.Models.Category;
using ZenMoney.Application.Models.PaymentMethod;
using ZenMoney.Application.Models.User;
using ZenMoney.Core.Enums;

namespace ZenMoney.Application.Models.Expense
{
    public class ExpenseModel : BaseModel
    {
        public ExpenseTypeEnum Type { get; set; }

        public DateOnly Date { get; set; }

        public string Description { get; set; }

        public decimal Amount { get; set; }

        public bool IsPaid { get; set; }

        public Guid CategoryId { get; set; }

        public CategoryModel Category { get; set; }

        public Guid PaymentMethodId { get; set; }

        public PaymentMethodModel PaymentMethod { get; set; }
    }
}

File: src/WebApp/ZenMoneyApp/src/app/responses/expense/expense-model.ts
import { ExpenseTypeEnum } from "../../enums/expense-type.enum";
import { BaseModel } from "../base.model";
import { CategoryModel } from "../category/category-model";
import { PaymentMethodModel } from "../payment-method/payment-method.model";

export interface ExpenseModel extends BaseModel {
    type: number;
    date: string;
    description: string;
    amount: number;
    isPaid: boolean;
    category: CategoryModel;
    paymentMethod: PaymentMethodModel;
}